docusaurus/docs/e2e-test-a-plugin/getting-started.md

```diff
@@ -0,0 +1,148 @@
+
+   Your Playwright config should have the following project configuration:
+
+<CodeSnippets
+snippets={[
+{ component: ScaffoldPluginE2EProjectsConfigNPM, label: 'npm' },
+{ component: ScaffoldPluginE2EProjectsConfigYarn, label: 'yarn' },
+{ component: ScaffoldPluginE2EProjectsConfigPNPM, label: 'pnpm' }
+]}
+groupId="package-manager"
+queryString="current-package-manager"
+/>
+
+### Step 3: Start Grafana
+
+Next, start up the latest version of Grafana on your local machine.
+
+<CodeSnippets
+snippets={[
+{ component: ScaffoldPluginE2EStartGrafanaNPM, label: 'npm' },
+{ component: ScaffoldPluginE2EStartGrafanaYarn, label: 'yarn' },
+{ component: ScaffoldPluginE2EStartGrafanaPNPM, label: 'pnpm' }
+]}
+groupId="package-manager"
+queryString="current-package-manager"
+/>
+
+If you want to start a specific version of Grafana, you can do that by specifying the `GRAFANA_VERSION` environment variable.
+
+```bash
+GRAFANA_VERSION=10.1.6 npm run server
+```
+
+### Step 4: Write tests
+
+In this example, we're using the panel edit page to test the query editor for a backend data source plugin. The plugin was scaffolded with the [create-plugin](https://github.com/grafana/plugin-tools/tree/main/packages/create-plugin) tool, and for this data source the query endpoint returns hard coded data points. This test asserts that the values `1` and `3` are being displayed in the `Table` panel.
+
+```ts title="queryEditor.spec.ts"
+import { test, expect } from '@grafana/plugin-e2e';
+
+test('data query should return values 1 and 3', async ({ panelEditPage, readProvisionedDataSource }) => {
+  const ds = await readProvisionedDataSource({ fileName: 'datasources.yaml' });
+  await panelEditPage.datasource.set(ds.name);
+  await panelEditPage.setVisualization('Table');
+  await expect(panelEditPage.refreshPanel()).toBeOK();
+  await expect(panelEditPage.panel.data).toContainText(['1', '3']);
+});
+```
+
+### Step 5: Run tests
+
+Now you open a new terminal and run the test script from within your local plugin development directory.
+
+<CodeSnippets
+snippets={[
+{ component: ScaffoldPluginE2ERunTestsNPM, label: 'npm' },
+{ component: ScaffoldPluginE2ERunTestsYarn, label: 'yarn' },
+{ component: ScaffoldPluginE2ERunTestsPNPM, label: 'pnpm' }
+]}
+groupId="package-manager"
+queryString="current-package-manager"
+/>
+
+### Step 6: Run tests in CI
+
+The following workflow can be used to run e2e tests against a matrix of Grafana versions for every PR in your Github repository. Note that this is a generic example based on a backend plugin. You may want to alter or remove a few of the steps in the `playwright-tests` job before using it in your plugin.
```

Comment:
```suggestion
We recommend updating your CI workflow to include the running of e2e tests to continuously check for breakages. The following workflow can be used to run e2e tests against a matrix of Grafana versions for every PR in your Github repository. Note that this is a generic example based on a backend plugin. You may want to alter or remove a few of the steps in the `playwright-tests` job before using it in your plugin.
```

Reply:
We need to explain the "matrix of grafana versions" more.

Reply:
Yeah let's just say `a range of Grafana versions`, and then we can elaborate on this in the full `CI` article

Reply:
sounds good